fastapi/routing.py

```diff
@@ -253,17 +253,16 @@ def __init__(
         name: str = None,
         dependency_overrides_provider: Any = None,
     ) -> None:
-        self.path = path
-        self.endpoint = endpoint
-        self.name = get_name(endpoint) if name is None else name
+        super().__init__(
+            path=path, endpoint=endpoint, name=name,
```

Comment:
Is there a reason for removing the current logic?
```suggestion
            path=path, endpoint=endpoint, name=get_name(endpoint) if name is None else name,
```
(the suggestion will probably make black unhappy)

Reply:
Case, when `name` argument is None, has already covered at the base class and it used the same `get_name ` function.

This is a code of a part of a `starllet.routing.WebSocketRoute` class:
```python
class WebSocketRoute(BaseRoute):
    def __init__(
        self, path: str, endpoint: typing.Callable, *, name: str = None
    ) -> None:
        assert path.startswith("/"), "Routed paths must start with '/'"
        self.path = path
        self.endpoint = endpoint
        self.name = get_name(endpoint) if name is None else name

        if inspect.isfunction(endpoint) or inspect.ismethod(endpoint):
            # Endpoint is function or method. Treat it as `func(websocket)`.
            self.app = websocket_session(endpoint)
        else:
            # Endpoint is a class. Treat it as ASGI.
            self.app = endpoint

        self.path_regex, self.path_format, self.param_convertors = compile_path(path)
```